lib/ktl/continous_reassigner.rb

```diff
@@ -0,0 +1,64 @@
+# encoding: utf-8
+
+module Ktl
+  class ContinousReassigner < Reassigner
+    include ZkClient::IZkDataListener
+
+    def initialize(zk_client, options={})
+      super(zk_client, options)
+      @latch = JavaConcurrent::CountDownLatch.new(1)
+      @sleeper = options[:sleeper] || java.lang.Thread
+      @delay = options[:delay] || 5
+      @shell = options[:shell]
+    end
+
+    def execute(reassignment)
+      Signal.trap('SIGINT', proc { @logger.info 'Exiting due to Ctrl-C'; @latch.count_down })
+      @zk_client.watch_data(zk_utils.class.reassign_partitions_path, self)
+      if reassignment_in_progress?
+        @logger.info 'reassignment already in progress, watching for changes...'
+        progress = ReassignmentProgress.new(@zk_client, logger: @logger, verbose: true)
+        progress.display(@shell)
+      else
+        reassign(reassignment)
+      end
+      @latch.await
```

Comment:
It's a bit weird to use a latch but keep the counting separate from the latch. Doesn't `reassignment` contain the number of things to wait for?